File: December 2024 Leetcode Solution/December-11.java
class Solution {
    public int maximumBeauty(int[] nums, int k) {
        int max_flower = Arrays.stream(nums).max().getAsInt();
        int max_range = max_flower + k * 2 + 1;
        int[] prefix_sum = new int[max_range + 1];

        for (int flower : nums) {
            prefix_sum[flower]++;
            if (flower + k * 2 + 1 <= max_range) {
                prefix_sum[flower + k * 2 + 1]--;
            }
        }

        int max_beauty = 0, running_sum = 0;
        for (int value : prefix_sum) {
            running_sum += value;
            max_beauty = Math.max(max_beauty, running_sum);
        }

        return max_beauty;
    }
}

File: December 2024 Leetcode Solution/December-11.py
from typing import List

class Solution:
    def maximumBeauty(self, flowers: List[int], steps: int) -> int:
        max_flower = max(flowers)
        max_range = max_flower + steps * 2 + 2
        prefix_sum = [0] * max_range

        for flower in flowers:
            prefix_sum[flower] += 1
            if flower + steps * 2 + 1 < max_range:
                prefix_sum[flower + steps * 2 + 1] -= 1

        max_beauty = running_sum = 0

        for value in prefix_sum:
            running_sum += value
            if running_sum > max_beauty:
                max_beauty = running_sum

        return max_beauty
